src/main/reactjs/src/index.tsx

```diff
@@ -1,5 +1,9 @@
 import ReactDOM from 'react-dom';
 
 import { App } from './App';
+import { initI18n } from 'configs/i18n';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+initI18n().then((_) => {
+  ReactDOM.render(<App />, document.getElementById('root'));
+  return;
+});
```

Comment:
Tästä aiheutuu pieni viive sovelluksen näyttämiseen: käännösten latauksen aikana näkyy vain spinneri.
Yksi mahdollisuus voisi olla että i18n initializen yhteydessä asetettaisiin vain nuo staattiset käännökset ja API-kutsujen kautta haettavat käännökset asetettaisiin erikseen blokkaamatta renderiä.

Reply:
Aivan, näin olikin tarkoitus, koska nyt se toimii oikein vaikka lokalisointi apin kutsu hidastelisi. Jotta voi tehdä kuten kuvasit, täytyy keksiä tapa jolla ui renderöidään uudelleen lokalisointien muuttumisen jälkeen.

Reply:
Dailyssä keskusteltiin vähän tästä. Voisi ehkä katsoa, että olisiko mitenkään mahdollista tehdä tämä build vaiheessa. Käännökset muuttuvat kuitenkin hyvin harvoin?

Reply:
Idea jatkojalostettavaksi: jos pidettäisiin koodistosta saatavat käännökset versionhallinnassa. Näitä käännöksiä voisi sitten vaikka ajastetusti github actionilla / botillla / tms. vertailla koodiston nykyhetkeä vasten. Jos löytyy eroja, avataan automaattisesti PR joka päivittäisi käännökset vastaamaan koodiston nykytilaa. Sopivasti tätä logiikkaa yleistämällä voisi olla hyödyllinen  kapistus useammallekin projektille.